### DISCUSSION TITLE
nomencl 宏包使用问题

### DISCUSSION THREAD

**Discussion by jwen15** [tuna/thuthesis#692] [Q&A] [answered]
## 编译环境
编译的系统：Windows 10
TeX 发行版：TeX Live 2021
模板版本：v7.2.3
模板类型：doctor

## 描述问题
#689 续
如果使用 `nomencl` 宏包，符号和缩略语说明那一页就没了。
我是用 TeXstudio 编译的，会报一个 warning：`No file main.nls`
需要我在 cmd.exe 中运行 `makeindex main.nlo -s nomencl.ist -o main.nls`，再重新编译，就一切正常了，并且可以自动排序了。

**Comment by zepinglee** [comment, marked as answer]
`nomencl` 的正常用法就是需要调用 `makeindex` 的。建议用 `latexmk` 编译，已经配置好了可以自动调用 `makeindex`。https://github.com/tuna/thuthesis/blob/1288c42a77c6d66e99c6e965948e38eeeeb531e5/latexmkrc#L14-L18

**Reply by jwen15** [reply]
知道了，谢谢！

**Comment by Dionmn** [comment]
我用Texstudio编译，生成了nls文件，报错：tocbasic.sty: 错误: 1377: Unknown message 'caption-code-by-class-deactivated' for module(LaTeX) 'tocbasic'. }，请问有人遇到过吗
